pandas/_libs/tslibs/period.pyx

```diff
@@ -2404,6 +2406,13 @@ class Period(_Period):
                 value = str(value)
             value = value.upper()
             dt, reso = parse_time_string(value, freq)
+            try:
+                ts = Timestamp(value, freq=freq)
+                nanosecond = ts.nanosecond
+                if nanosecond != 0:
+                    reso = 'nanosecond'
+            except (ValueError, OutOfBoundsDatetime):
```

Comment:
OOBDatetime is a subclass of ValueError so this is uneeded

Reply:
I've first try to only use `OutOfBoundsDatetime` but the test code fails on many VM. Here is an example: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=37155&view=logs&j=a3a13ea8-7cf0-5bdb-71bb-6ac8830ae35c&t=add65f64-6c25-5783-8fd6-d9aa1b63d9d4&l=471

<details>

```python
pandas/_libs/tslibs/period.pyx:2410: in pandas._libs.tslibs.period.Period.__new__
    ts = Timestamp(value, freq=freq)
pandas/_libs/tslibs/timestamps.pyx:848: in pandas._libs.tslibs.timestamps.Timestamp.__new__
    ts = convert_to_tsobject(ts_input, tz, unit, 0, 0, nanosecond or 0)
pandas/_libs/tslibs/conversion.pyx:333: in pandas._libs.tslibs.conversion.convert_to_tsobject
    return convert_str_to_tsobject(ts, tz, unit, dayfirst, yearfirst)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   raise ValueError("could not convert string to Timestamp")
E   ValueError: could not convert string to Timestamp

pandas/_libs/tslibs/conversion.pyx:581: ValueError
```

</details>

Reply:
@OlivierLuG I think Jeff meant that it should be enough to only use `ValueError`, as that would already cover the `OutOfBoundsDatetime` case

Reply:
OK, I got it. I will try a new PR with only `ValueError`